studiopress/pattern-manager
# Get list of pattern categories via pattern category registry (Patterns view)

Currently, the list of categories for the `app` module is parsed from pattern files. Since categories in the pattern header are saved by their `name`, parsing from the pattern file can be problematic if the category was registered with a `name` that differs from its `label`.

This PR hydrates the `app` with a list of registered categories, populated via call to the pattern category registry.

Fixes [GF-3706](https://wpengine.atlassian.net/browse/GF-3706).

---

For example, here is the current way, using Frost as reference:

![Screenshot 2023-03-08 at 10 06 37 AM](https://user-images.githubusercontent.com/108079556/223767432-89c20fb8-ccff-47fd-bd72-5b2da8d340f4.png)

The displayed values are all parsed directly from the `name` stored in pattern files, with dashes removed.

---

Here is the same categories list with `label` values mapped against the categories registry:

![Screenshot 2023-03-08 at 10 06 22 AM](https://user-images.githubusercontent.com/108079556/223768245-23ebda7a-6987-4bd4-8bab-cdf8f0f5e036.png)

The displayed values are the actual registered `label` values registered by Frost.

---

### How to test

1. Checkout the branch
2. With Frost active, navigate to the Patterns view
3. The categories list should be populated with registered `label` values

---

### Notes
Hydrating the `app` with the list of `patternCategories` will also be required for bulk assignment of categories to patterns, so having this data available now will be a useful head start for that work.
